walk-in-interview/src/main/webapp/account/applicant/edit/script.js

```diff
@@ -0,0 +1,216 @@
+  skillsTitle.innerText = STRINGS['skills-title'];
+
+  renderSkillsList(applicantDetails.skills);
+}
+
+/**
+ * Gets the applicantId from the url.
+ * @return {String} The applicantId.
+ */
+function getApplicantId() {
+  const queryString = window.location.search;
+
+  const urlParams = new URLSearchParams(queryString);
+  if (urlParams === '') {
+    throw new Error('url params should not be empty');
+  }
+
+  return urlParams.get(APPLICANT_ID_PARAM);
+}
+
+/**
+ * Gets the applicant details from the url.
+ * @return {JSON} The applicant detail json.
+ */
+function getApplicantDetails() {
+  const queryString = window.location.search;
+
+  const urlParams = new URLSearchParams(queryString);
+  if (urlParams === '') {
+    throw new Error('url params should not be empty');
+  }
+
+  return JSON.parse(urlParams.get(APPLICANT_DETAILS_PARAM));
+}
+
+/**
+ * Renders the skill list with existing skills prefilled.
+ * 
+ * @param {Array} skills Existing skills.
+ */
+function renderSkillsList(skills) {
+  const fullSkillsMap = getRequirementsList();
+
+  const skillsListElement = document.getElementById('skills');
+
+  // Resets the list in case renders the same requirements twice
+  skillsListElement.innerHTML = '';
+  const skillElementTemplate =
+    document.getElementById('skill-element-template');
+
+  for (const key in fullSkillsMap) {
+    if (!fullSkillsMap.hasOwnProperty(key)) {
+      continue;
+    }
+
+    const skillElement = skillElementTemplate
+        .cloneNode(/* includes child elements */ true);
+
+    // tick box
+    const checkbox = skillElement.children[0];
+    checkbox.setAttribute('id', key);
+    checkbox.setAttribute('value', key);
+    checkbox.setAttribute('name', 'skill');
+
+    // If this skill is one of the existing for this applicant, tick it
+    if (skills.includes(fullSkillsMap[key])) {
+      checkbox.setAttribute('checked', true);
+    }
+
+    // text label
+    const label = skillElement.children[1];
+    label.setAttribute('for', key);
+    label.innerHTML = fullSkillsMap[key];
+
+    skillsListElement.appendChild(skillElement);
+  }
+}
+
+function getApplicantDetailsFromUserInput() {
+  const applicantId = getApplicantId();
+  const applicantName = document.getElementById('name').value;
+
+  const skillsCheckboxes =
+    document.getElementsByName('skill');
+  const skillsList = [];
+  skillsCheckboxes.forEach(({checked, id}) => {
+    if (checked) {
+      skillsList.push(id);
+    }
+  });
+
+  const applicantDetails = {
+    accountId: applicantId,
+    name: applicantName,
+    skills: skillsList,
+  };
+
+  return applicantDetails;
+}
+
+/**
+* Validates the user input.
+* Shows error message on the webpage if there is field with invalid input.
+*
+* @return {boolean} depending on whether the input is valid or not.
+*/
+function validateRequiredUserInput() {
+  const applicantId = getApplicantId();
+  const name = document.getElementById('name').value;
+
+  if (applicantId === '') {
```

Comment:
Do user enter applicant id? Maybe this is something wrong rather than invalid user input?